ext/standard/uniqid.c

```diff
@@ -25,48 +26,53 @@
 #include <errno.h>
 
 #include <stdio.h>
-#ifdef PHP_WIN32
-#include "win32/time.h"
-#else
-#include <sys/time.h>
-#endif
 
 #include "ext/random/php_random.h"
 #include "ext/random/php_random_csprng.h"
 
-#ifdef HAVE_GETTIMEOFDAY
-ZEND_TLS struct timeval prev_tv = { 0, 0 };
+ZEND_TLS time_t prev_sec  = 0;
+ZEND_TLS long   prev_usec = 0;
 
 /* {{{ Generates a unique ID */
 PHP_FUNCTION(uniqid)
 {
 	char *prefix = "";
 	bool more_entropy = 0;
 	zend_string *uniqid;
-	int sec, usec;
 	size_t prefix_len = 0;
-	struct timeval tv;
+	time_t sec;
+	long nsec, usec;
+	int isec, iusec;
 
 	ZEND_PARSE_PARAMETERS_START(0, 2)
 		Z_PARAM_OPTIONAL
 		Z_PARAM_STRING(prefix, prefix_len)
 		Z_PARAM_BOOL(more_entropy)
 	ZEND_PARSE_PARAMETERS_END();
 
-	/* This implementation needs current microsecond to change,
-	 * hence we poll time until it does. This is much faster than
-	 * calling usleep(1) which may cause the kernel to schedule
-	 * another process, causing a pause of around 10ms.
+	/* This implementation needs current microsecond to change.
+	 * We will manually move the time forward if needed. This is
+	 * more stable as polling or sleeping as it will not be
+	 * effected by time changes (like ntp) and this is much faster
+	 * than calling usleep(1) which may cause the kernel to
+	 * schedule another process, causing a pause of around 10ms.
 	 */
```

Comment:
This should not include functional changes as part of the refactoring.

Reply:
I have separated this part out into it's own PR: https://github.com/php/php-src/pull/19264